src/js/common/schemaform/save-in-progress/RoutedSavableReviewPage.jsx

```diff
@@ -0,0 +1,136 @@
+  constructor(props) {
+    super(props);
+    this.autoSave = this.autoSave.bind(this);
+    this.debouncedAutoSave = _.debounce(1000, this.autoSave);
+  }
+
+  setData = (...args) => {
+    this.props.setData(...args);
+    this.debouncedAutoSave();
+  }
+
+  autoSave() {
+    const { form, user } = this.props;
+    if (user.login.currentlyLoggedIn) {
+      const data = form.data;
+      const { formId, version } = form;
+      const returnUrl = this.props.location.pathname;
+
+      this.props.autoSaveForm(formId, data, version, returnUrl);
+    }
+  }
+
+  renderErrorMessage = () => {
+    const { route, user, form, location } = this.props;
+    const errorText = route.formConfig.errorText;
+    const savedStatus = form.savedStatus;
+
+    const saveLink = (<SaveFormLink
+      locationPathname={location.pathname}
+      form={form}
+      user={user}
+      saveAndRedirectToReturnUrl={this.props.saveAndRedirectToReturnUrl}
+      toggleLoginModal={this.props.toggleLoginModal}>
+      save your application
+    </SaveFormLink>);
+
+    if (saveErrors.has(savedStatus)) {
+      return saveLink;
+    }
+
+    let InlineErrorComponent;
+    if (typeof errorText === 'function') {
+      InlineErrorComponent = errorText;
+    } else if (typeof errorText === 'string') {
+      InlineErrorComponent = () => <p>{errorText}</p>;
+    } else {
+      InlineErrorComponent = () => <p>If it still doesn’t work, please call the Vets.gov Help Desk at <a href="tel:855-574-7286">1-855-574-7286</a>, TTY: <a href="tel:18008778339">1-800-877-8339</a>. We’re here Monday &#8211; Friday, 8:00 a.m. &#8211; 8:00 p.m. (ET).</p>;
+    }
+
+    return (
+      <div className="usa-alert usa-alert-error schemaform-failure-alert">
+        <div className="usa-alert-body">
+          <p className="schemaform-warning-header"><strong>We’re sorry, the application didn’t go through.</strong></p>
+          <p>We’re working to fix the problem, but it may take us a little while. Please {saveLink} and try submitting it again tomorrow.</p>
+          {!user.login.currentlyLoggedIn && <p>If you don’t have an account, you’ll have to start over. Please try submitting your application again tomorrow.</p>}
+          <InlineErrorComponent/>
+        </div>
+      </div>
+    );
+  }
+
+  render() {
+    const { form, user, location } = this.props;
+
+    const contentAfterButtons = (
+      <div>
+        <SaveStatus
+          isLoggedIn={user.login.currentlyLoggedIn}
+          showLoginModal={user.login.showModal}
+          toggleLoginModal={this.props.toggleLoginModal}
+          form={form}>
+        </SaveStatus>
+        <SaveFormLink
+          locationPathname={location.pathname}
+          form={form}
+          user={user}
+          saveAndRedirectToReturnUrl={this.props.saveAndRedirectToReturnUrl}
+          toggleLoginModal={this.props.toggleLoginModal}/>
+      </div>
+    );
+
+    return (
+      <ReviewPage
+        {...this.props}
```

Comment:
This will also pass all the props from `connect`, which `ReviewPage` won't need. `user` shouldn't cause any conflicts (just extra prop passing), but I'm not sure how `form` will interact with the`form` from `ReviewPage`'s `mapStateToProps`.

If there's no conflict by passing `form`, the worst I can think of that'll happen is unnecessary renders because `user` changed in the redux store, but I figured it's probably worth mentioning.

Reply:
This page is not using the connected `ReviewPage`, but the regular component. So it should conflict. But you're right that it is passing some properties that are unnecessary. I went back and forth on deciding if it was less brittle to do this vs passing them individually (which is more than just form, it also includes the action functions).

Reply:
Ohhh, I see what you did there. Clever! 😄 
Yeah, like I said, worst case scenario is it updates more than it should because props `ReviewPage` doesn't care about are being changed. `RoutedSavableReviewPage` is updating because of those props anyhow, so...kinda whatever, I guess.